src/fontra/views/editor/panel-transformation.js

```diff
@@ -450,11 +581,266 @@ export default class TransformationPanel extends Panel {
+    );
+
+    const editBehavior = behaviorFactory.getBehavior("default");
+    const editChange = editBehavior.makeChangeForDelta(delta);
+    return [editChange, editBehavior.rollbackChange];
+  }
+}
+
+// Define moveDescriptor objects
+class AlignObjectsDescriptor {
+  constructor(descriptor, position, directionVar) {
+    this.position = position;
+    this.undoLabel = `align ${descriptor}`;
+    this.minProperty = `${directionVar}Min`;
+    this.maxProperty = `${directionVar}Max`;
+    this.property = `${directionVar}${position}`;
+    this.deltaProperty = directionVar;
+  }
+
+  computeDeltasFromBoundingBoxes(boundingBoxes) {
+    const maxes = boundingBoxes.map((bounds) => bounds[this.maxProperty]);
+    const mins = boundingBoxes.map((bounds) => bounds[this.minProperty]);
+    let maximum = Math.max(...maxes);
+    let minimum = Math.min(...mins);
+    let deltas = [];
+    for (const bounds of boundingBoxes) {
+      const delta = { x: 0, y: 0 };
+      if (this.position === "Center") {
+        delta[this.deltaProperty] =
+          maximum -
+          bounds[this.maxProperty] +
+          (bounds[this.maxProperty] - bounds[this.minProperty]) / 2 -
+          (maximum - minimum) / 2;
+      } else {
+        const pos = this.position === "Min" ? minimum : maximum;
+        delta[this.deltaProperty] = pos - bounds[this.property];
+      }
+      deltas.push(delta);
+    }
+    return deltas;
+  }
+}
+
+const alignBottom = new AlignObjectsDescriptor("bottom", "Min", "y");
+const alignTop = new AlignObjectsDescriptor("top", "Max", "y");
+const alignLeft = new AlignObjectsDescriptor("left", "Min", "x");
+const alignRight = new AlignObjectsDescriptor("right", "Max", "x");
+const alignMiddle = new AlignObjectsDescriptor("middle", "Center", "y");
+const alignCenter = new AlignObjectsDescriptor("center", "Center", "x");
+
+class DistributeObjectsDescriptor {
+  constructor(direction, directionVar) {
+    this.undoLabel = `distribute ${direction}`;
+    this.minProperty = `${directionVar}Min`;
+    this.maxProperty = `${directionVar}Max`;
+    this.deltaProperty = directionVar;
+  }
+
+  computeDeltasFromBoundingBoxes(boundingBoxes, distributeValue) {
+    let effectiveSize = 0;
+    for (const bounds of boundingBoxes) {
+      effectiveSize += bounds[this.maxProperty] - bounds[this.minProperty];
+    }
+    const mins = boundingBoxes.map((bounds) => bounds[this.minProperty]);
+    const maxes = boundingBoxes.map((bounds) => bounds[this.maxProperty]);
+    const minimum = Math.min(...mins);
+    const maximum = Math.max(...maxes);
+
+    let distributionSpacing =
+      (maximum - minimum - effectiveSize) / (boundingBoxes.length - 1);
+    if (!isNaN(distributeValue)) {
+      distributionSpacing = distributeValue;
+    }
+
+    let next = minimum;
+    let deltas = [];
+    for (const bounds of boundingBoxes) {
+      const dimentions = bounds[this.maxProperty] - bounds[this.minProperty];
```

Comment:
typo: dimensions. But I think this again is the wrong name. Consider `size`. Although `extent` is a more precise term.

Reply:
Thanks. done.